### PR TITLE
Improve some wording in disk caching docs

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Improve some wording in disk caching docs

## Motivation and Context
cleanup

## How Has This Been Tested?
visually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/disk-caching/DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Disk caching design [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
```
Comment (nitisht) [inline docs/disk-caching/DESIGN.md]: We follow the TitleCase for doc headers, suggest we do it here as well. This should be `Disk Caching Design`

File: docs/disk-caching/DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- Caches new objects for entries not found in cache while downloading. Otherwise serves from the cache.
+- Caches all successfully uploaded objects. Replaces existing cached entry of the same object if needed.
+- When an object is deleted, corresponding entry in cache if any is deleted as well.
+- Cache continues to work for read-only operations such as GET, HEAD when backend is offline.
+- Cache disallows write operations when backend is offline.
+
+> NOTE: Expiration happens automatically based on the configured interval as explained above, frequently accessed objects stay alive in cache for a significantly longer time.
+
+## Limits
+- Bucket policies are not cached, so anonymous operations are not supported when backend is offline.p
```
Comment (poornas) [inline docs/disk-caching/DESIGN.md]: typo after offline. remove extraneous p

File: docs/disk-caching/DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+- Objects are distributed using deterministic hashing among the list of configured cache drives. If one or more drives go offline, or cache drive configuration is altered in any way. Performance may degrade to a linear lookup time depending on the number of disks in cache.
```
Comment (poornas) [inline docs/disk-caching/DESIGN.md]: If one or more drives go offline, or cache drive configuration is altered in any way. Performance may degrade -> If one or more drives go offline, or cache drive configuration is altered in any way, performance may degrade

**Review by poornas — CHANGES_REQUESTED** [review]

**Review by nitisht — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Pull request merged**